src/jdk.attach/linux/classes/sun/tools/attach/VirtualMachineImpl.java

```diff
@@ -237,40 +192,19 @@ InputStream execute(String cmd, Object ... args) throws AgentLoadException, IOEx
     /*
      * InputStream for the socket connection to get target VM
      */
-    private static class SocketInputStream extends InputStream {
-        int s = -1;
-
-        public SocketInputStream(int s) {
-            this.s = s;
+    private static class SocketInputStreamImpl extends SocketInputStream {
```

Comment:
Can this class definition also be shared by making it a protected nested class in the superclass?

Reply:
Sorry I see it already is, but I think we can do better.

Reply:
This(`SocketInputStreamImpl`) is implementation-specific, it calls platform-dependent VirtualMachine.xxx.